client/me/purchases/remove-purchase/index.jsx

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Dialog from 'components/dialog';
+import { getPurchase, isDataLoading } from '../utils';
+import { getName, isRemovable } from 'lib/purchases';
+import Gridicon from 'components/gridicon';
+import notices from 'notices';
+import purchasePaths from '../paths';
+import { removePurchase } from 'lib/upgrades/actions';
+
+const RemovePurchase = React.createClass( {
+	propTypes: {
+		selectedPurchase: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired
+	},
+
+	getInitialState() {
+		return {
+			isDialogVisible: false,
+			isRemoving: false
+		};
+	},
+
+	closeDialog() {
+		this.setState( { isDialogVisible: false } );
+	},
+
+	openDialog( event ) {
+		event.preventDefault();
+
+		this.setState( { isDialogVisible: true } );
+	},
+
+	removePurchase( closeDialog ) {
+		this.setState( { isRemoving: true } );
+
+		removePurchase( this.props.selectedPurchase.data.id, this.props.user.ID, success => {
+			this.setState( { isRemoving: false } );
+
+			if ( success ) {
+				notices.success(
+					this.translate( `The purchase was removed from {{em}}%(siteSlug)s{{/em}}.`, {
+						args: { siteSlug: this.props.selectedSite.slug },
+						components: { em: <em /> }
+					} ),
+					{ persistent: true }
+				);
+
+				page( purchasePaths.list() );
+			} else {
+				closeDialog();
+
+				notices.error( this.props.selectedPurchase.error );
+			}
+		} );
+	},
+
+	renderCard() {
+		const productName = getName( getPurchase( this.props ) );
+
+		return (
+			<CompactCard className="remove-purchase__card" onClick={ this.openDialog }>
+				<a >
```

Comment:
The problem with using an `href`attribute in this anchor is that the cursor will only change if you hover the link itself whereas we probably want to change it whenever the cursor enters the card (especially since the `onclick` event applies to the whole card). So `cursor: pointer;` would be needed anyway (@ebinnion or @lezama: that's probably something you want to fix in the `<DeleteUser />` component).

Note I had to inject this anchor in this `CompactCard` because I wasn't able to use its `href` prop in a way which would give me similar results.